src/Grids/Grids.jl

```diff
@@ -80,19 +80,27 @@ Abstract supertype for grids with elements of type `FT` and topology `{TX, TY, T
 """
 abstract type AbstractGrid{FT, TX, TY, TZ} end
 
+"""
+    AbstractPrimaryGrid{FT, TX, TY, TZ}
+
+Abstract supertype for "primary" grids (as opposed to grids with immersed boundaries)
+with elements of type `FT` and topology `{TX, TY, TZ}`.
```

Comment:
Out of curiosity is _primary grid_ a common term in the immersed boundary literature?

Reply:
I don't think so. Another term might be "underlying grid". Not sure what's best...

Reply:
If it's just an internal abstract type then what do you think of `AbstractUnimmersedGrid`? It's is a little verbose but might be clearer?

Reply:
I guess `Primary` or `Underlying` generalizes to other grid wrapper types, but `Unimmersed` would have to be refactored if we had another wrapper type, possibly creating more work for us (I'm not sure how generalized vertical coordinates will be implemented, but it might be via this method).

The other issue is that we have an immersed boundary implementation which would then use `AbstractUnimmersedGrid`, which might be a little confusing, but I'm not sure.

Reply:
Hmmm so I guess `AbstractUnimmersedGrid` makes sense for this PR but maybe not for future work.

I find `Primary` and `Underlying` kinda vague but the name is explained in the docstring and it shouldn't be hard to change if we come up with a better name.